src/components/KPop/KPop.vue

```diff
@@ -235,8 +237,25 @@ const popoverClassesObj = computed(() => [props.popoverClasses, { 'hide-caret':
 const popoverPlacement = computed((): PopPlacements => props.positionFixed ? props.placement : 'top')
 
 const { floatingStyles, placement: calculatedPlacement, update: updatePosition } = useFloating(popoverTrigger, popoverElement, {
-  ...(popoverPlacement.value === 'auto' ? { middleware: [autoPlacement()] } : { placement: popoverPlacement.value, middleware: [flip(), shift()] }),
+  ...(popoverPlacement.value === 'auto' && { middleware: [autoPlacement()] }), // when placement is auto just use autoPlacement middleware
+  ...(popoverPlacement.value !== 'auto' && {
+    placement: popoverPlacement.value,
+    middleware: [
+      flip(),
+      shift(),
+      size({
+        apply({ elements, availableHeight }) {
+          requestAnimationFrame(() => {
+            Object.assign(elements.floating.style, {
+              maxHeight: `${availableHeight}px`,
+            })
+          })
```

Comment:
nit: can you add a comment explaining this?

Reply:
I did

![Screenshot 2024-06-06 at 2 41 45 PM](https://github.com/Kong/kongponents/assets/36751160/8e58305c-1521-4bbc-beb9-abab66f1b590)

Reply:
lol whoops